script.py

```diff
@@ -66,22 +69,30 @@ def get_regions(self):
         return sorted(regions)
 
     def get_reserved_instances_offerings(self, region):
-        response = self.get_ec2_client(region).describe_reserved_instances_offerings(
-            IncludeMarketplace=False,
-            InstanceType=INSTANCE_TYPE,
-            InstanceTenancy=INSTANCE_TENANCY,
-            MinDuration=DURATION,
-            MaxDuration=DURATION,
-            OfferingClass=OFFERING_CLASS,
-            OfferingType=OFFERING_TYPE,
-            ProductDescription=PRODUCT_DESCRIPTION,
-        )
-
-        return {
-            offering['ReservedInstancesOfferingId']: offering['AvailabilityZone'][-1:]
-            for offering in response['ReservedInstancesOfferings']
-            if 'AvailabilityZone' in offering
-        }
+        if self.name in bad_credential_list:
```

Comment:
@ekuns Could you replace the `bad_credential_list` with a property like `self.valid_credentials` which is set to `True` in `__init__` and sets it to `False` when it catches a `NoCredentialsError`?

Reply:
Done!  I'm still learning Python so my code isn't always very Pythonic.  Your asking me to make the change helps me get better at writing Python the correct way, so thanks for that!